### PR TITLE
[BUU] Sticky table header and saving banner - alternate solution

### LINKED ISSUE #11902: [BUU] Sticky table header and saving banner
**Scenario 1: Scrolling - sticky table header**
Given that I have loaded my product list
When I scroll below the fold
Then the header of the table remains visible
And I can see it as I keep scrolling

**Scenario 2: Scrolling - sticky message when products have been modified**
Given that I have loaded my product list
And I have edited one or more products
And the message saying that products have been edited is displayed
When I scroll below the fold
Then the message remains visible
And I can see it as I keep scrolling

## Design specs

Prototype for scrolling here > https://www.figma.com/proto/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery---Back-Office?page-id=470%3A5243&type=design&node-id=470-5244&viewport=628%2C475%2C0.87&t=hV7Mk4xqjzBy5uwU-1&scaling=min-zoom&mode=design

sticky behaviour with saving banner here: https://www.figma.com/proto/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery---Back-Office?page-id=363%3A2039&type=design&node-id=392-2584&viewport=282%2C245%2C0.07&t=R0ivzyZuaZOooAHl-1&scaling=min-zoom&starting-point-node-id=392%3A2584&show-proto-sidebar=1&mode=design

### PR DESCRIPTION
**Pull request by dacook**
#### What? Why?

- Closes #11902
- Closes https://github.com/openfoodfoundation/openfoodnetwork/pull/11935 as alternative solution.

#### What should we test?
As per above issue:

- With admin_style_v3 feature enabled, visit `/admin/products` page

**Scenario 1: Scrolling - sticky table header**
Given that I have loaded my product list
When I scroll below the fold
Then the header of the table remains visible
And I can see it as I keep scrolling
Screenshot 2024-01-10 at 12 37 46 pm

**Scenario 2: Scrolling - sticky message when products have been modified**
Given that I have loaded my product list
And I have edited one or more products
And the message saying that products have been edited is displayed
When I scroll below the fold
Then the message remains visible
And I can see it as I keep scrolling
Screenshot 2024-01-10 at 12 38 34 pm

**Scenario 3: Scrolling - sticky message when product save failed, and other products have been modified**
Given that I have loaded my product list
And I have edited one or more products with invalid value (eg empty name)
And I clicked "Save changes"
And then I edited another product.
The message has multiple lines.
When I scroll below the fold
Then the message remains visible
And I can see it as I keep scrolling

Screenshot 2024-01-10 at 12 38 55 pm
Screenshot 2024-01-10 at 12 39 17 pm

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [x] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by dacook** [comment]
Work in progress: the "discard" and "Save" button colours need fixing up.

File: app/webpacker/css/admin/products_v3.scss [hunk 1]
```diff
@@ -36,12 +39,38 @@
...
     border-collapse: separate; // This is needed for the outer padding. Also should be helpful to give more flexibility of borders between rows.
 
     // Additional horizontal padding to align with input contents
-    thead th.with-input {
-      padding-left: $padding-tbl-cell + $hpadding-txt;
-      padding-right: $padding-tbl-cell + $hpadding-txt;
+    thead {
+      position: sticky;
+      top: 0;
+      z-index: 1; // TODO: Cover .popout and .vertical-ellipsis-menu, but only when sticky
```
Comment (rioug) [inline app/webpacker/css/admin/products_v3.scss:49]: Shouldn't this use variable ?

**Review by rioug — APPROVED** [review]
Looks good !

File: app/webpacker/css/admin/products_v3.scss (hunk 1 above)
Comment (dacook) [reply app/webpacker/css/admin/products_v3.scss:49]: Yes. I was expecting to come back and do something more advanced here, but haven't so far. 
Hmm I will add it 👍

Comment (dacook) [reply app/webpacker/css/admin/products_v3.scss:49]: Changed my mind: I reckon there's no need to register it when z-index is `1`. For anything higher, we should register a global variable.

**Review by mkllnk — APPROVED** [review]
Great solution.

**Comment by RachL** [comment]
LGTM! Merging :)

**Pull request merged**